src/transformers/trainer.py

```diff
@@ -1975,8 +1985,20 @@ def zero_grad(x):
                     xp.trace_detached('127.0.0.1:9012', profile_logdir or tempfile.mkdtemp(), profile_duration or 20000)
 
                 if self.control.should_epoch_stop or self.control.should_training_stop:
+                    # PyTorch/XLA relies on the data loader to insert the mark_step for
```

Comment:
What's this?

Reply:
Upstream added this https://github.com/huggingface/transformers/blob/91d155ea92da372b319a79dd4eef69533ee15170/src/transformers/trainer.py#L2286-L2290